substrate/client/db/src/lib.rs

```diff
@@ -1693,19 +1693,19 @@ impl<Block: BlockT> Backend<Block> {
 							number,
 							hash,
 						)?;
-					}
-					if start > end {
-						transaction.remove(columns::META, meta_keys::BLOCK_GAP);
-						block_gap = None;
-						debug!(target: "db", "Removed block gap.");
-					} else {
-						block_gap = Some((start, end));
-						debug!(target: "db", "Update block gap. {:?}", block_gap);
-						transaction.set(
-							columns::META,
-							meta_keys::BLOCK_GAP,
-							&(start, end).encode(),
-						);
+						if start > end {
```

Comment:
Are you sure the block gap can't be updated externally and returned from `self.blockchain.meta.read()`?

Reply:
Yes, I'm pretty sure that `update_block_gap()` is the only way to update the gap in meta and it's only invoked at the end of `try_commit_operation()`.